### PR TITLE
Adds transition overlay

### LINKED ISSUE #11333: Notification not shown after transition was triggered
## Test plan

https://github.com/brave/browser-laptop/pull/11395#issue-263840000

----

### Description
For the BAT Mercury launch we are planning to show a notification that the funds get transferred from BTC to BAT. 

https://github.com/brave/browser-laptop/pull/11231 delivered most of this- the only problem is the notification is not shown.  The state needed to show the notification is only triggered after the transition has started... however the only place where these are checked is during launch

### Steps to Reproduce
- Have a bitcoin wallet (ex: use 0.18.x)
- Launch a BAT Mercury build using `LEDGER_VERBOSE=true` on command line
- Verify transition is triggered (this starts the process; it takes a while)
- Notice you are not shown alert

### LINKED ISSUE #11338: Currency QR Code missing after upgrade
## Test plan

https://github.com/brave/browser-laptop/pull/11395#issue-263840000

----

### Description
Currency QR Code missing after upgrade

### Steps to Reproduce

  1. Set browser to 0.18.36 and enable payments
  2. Upgrade to 0.19.34
  3. Click add funds in payment
  4. Click on currencies button, QR code is missing for all available currencies

**Actual result:**

image

**Expected result:**
Should show the QR code 
**Reproduces how often:** [What percentage of the time does it reproduce?]

### Brave Version

**about:brave info:**

Brave | 0.19.34
-- | --
rev | d87b7ca
Muon | 4.4.25

**Reproducible on current live release:**

N/A
### Additional Information

cc: @NejcZdovc

### LINKED ISSUE #11340: Disable backup wallet button until wallet is ready
## Test plan

https://github.com/brave/browser-laptop/pull/11395#issue-263840000

----

### Description

Disable backup wallet button until wallet is ready
### Steps to Reproduce

  1. Upgrade to 0.19.34
  2. Open payments
  3. Before wallet is converted, click on advance setting and click on back up wallet 
  4. No code words are shown

**Actual result:**

image

**Expected result:**
Should show an appropriate message or disable the button all together

**Reproduces how often:** [What percentage of the time does it reproduce?]

### Brave Version

**about:brave info:**

Brave | 0.19.34
-- | --
rev | d87b7ca
Muon | 4.4.25

**Reproducible on current live release:**

N/A
### Additional Information

cc: @evq @NejcZdovc

### LINKED ISSUE #11389: Progress screen while in transition
## Test plan

https://github.com/brave/browser-laptop/pull/11395#issue-263840000

----

### Description

Add overlay to brave payments until transition is done

### PR DESCRIPTION
**Pull request by NejcZdovc**
Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Resolves #11389
Resolves #11340
Resolves #11333 
Resolves #11338

Auditors:

Test Plan:

1. Create wallet with 0.18 version
2. With this profile switch to version 0.19
3. You should see transition overlay with additional settings disabled
4. After transition is done, overlay should hide and additional settings should be re-enabled

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: app/renderer/components/preferences/payment/enabledContent.js [hunk 1]
```diff
@@ -310,6 +337,75 @@ const styles = StyleSheet.create({
...
+    height: '15px',
+    borderRadius: '15px'
+  },
+
+  loader__line_1: {
+    backgroundColor: '#FF5000',
+    animationName: [loaderAnimation],
+    animationDuration: '.6s',
+    animationDelay: '.1s',
+    animationIterationCount: 'infinite'
```
Comment (luixxiul) [inline app/renderer/components/preferences/payment/enabledContent.js]: please don't forget to dedupe the animation properties later

File: app/renderer/components/preferences/payment/enabledContent.js [hunk 2]
```diff
@@ -171,8 +171,23 @@ class EnabledContent extends ImmutableComponent {
   render () {
     const ledgerData = this.props.ledgerData
     const walletStatusText = walletStatus(ledgerData)
-
-    return <section>
+    const inTranstion = !(ledgerData.getIn(['migration', 'btc2BatTransitionDone']))
+
+    return <section className={css(styles.enabledContent)}>
+      <div className={css(styles.loader, inTranstion && styles.loader_show)}>
```
Comment (luixxiul) [inline app/renderer/components/preferences/payment/enabledContent.js]: `styles.enabledContent__loader, inTranstion && styles.enabledContent__loader_show`. please change elements' className accordingly.

File: app/renderer/components/preferences/payment/enabledContent.js [hunk 3]
```diff
@@ -171,8 +171,23 @@ class EnabledContent extends ImmutableComponent {
...
+        </p>
+        <div className={css(styles.leader__wrap)}>
+          <div>
+            <div className={css(styles.loader__line, styles.loader__line_1)} />
+            <div className={css(styles.loader__line, styles.loader__line_2)} />
+            <div className={css(styles.loader__line, styles.loader__line_3)} />
+          </div>
+        </div>
+      </div>
       <div className={css(styles.walletBar)} data-test-id='walletBar'>
```
Comment (luixxiul) [inline app/renderer/components/preferences/payment/enabledContent.js]: `styles.walletBar` -> `styles.enabledContent__walletBar`

File: app/renderer/components/preferences/payment/enabledContent.js [hunk 4]
```diff
@@ -274,6 +289,18 @@ const gridStyles = StyleSheet.create({
   }
 })
 
+const loaderAnimation = {
```
Comment (cezaraugusto) [inline app/renderer/components/preferences/payment/enabledContent.js]: I know still WIP but please move this to `animations.js` file which is avail in 0.19.x too

File: app/renderer/components/preferences/payment/enabledContent.js (hunk 2 above)
Comment (NejcZdovc) [reply app/renderer/components/preferences/payment/enabledContent.js]: done

File: app/renderer/components/preferences/payment/enabledContent.js (hunk 3 above)
Comment (NejcZdovc) [reply app/renderer/components/preferences/payment/enabledContent.js]: this is not part of this PR

File: app/renderer/components/preferences/payment/enabledContent.js (hunk 4 above)
Comment (NejcZdovc) [reply app/renderer/components/preferences/payment/enabledContent.js]: done

File: app/renderer/components/preferences/payment/enabledContent.js (hunk 1 above)
Comment (NejcZdovc) [reply app/renderer/components/preferences/payment/enabledContent.js]: done

File: app/renderer/components/preferences/payment/enabledContent.js [hunk 5]
```diff
@@ -171,8 +170,22 @@ class EnabledContent extends ImmutableComponent {
...
+        </p>
+        <div className={css(styles.leader__wrap)}>
+          <div>
+            <div className={css(styles.loader__line, styles.loader__line_1, !inTranstion && styles.loader__line_off)} />
+            <div className={css(styles.loader__line, styles.loader__line_2, !inTranstion && styles.loader__line_off)} />
+            <div className={css(styles.loader__line, styles.loader__line_3, !inTranstion && styles.loader__line_off)} />
+          </div>
+        </div>
+      </div>
       <div className={css(styles.walletBar)} data-test-id='walletBar'>
```
Comment (luixxiul) [inline app/renderer/components/preferences/payment/enabledContent.js]: `styles.walletBar` -> `styles.enabledContent__walletBar`. See L175.

Comment (NejcZdovc) [reply app/renderer/components/preferences/payment/enabledContent.js]: done

File: app/renderer/components/preferences/payment/enabledContent.js (hunk 1 above)
Comment (luixxiul) [reply app/renderer/components/preferences/payment/enabledContent.js]:
````
loader__line: {
     display: 'inline-block',
     width: '15px',
     height: '15px',
     borderRadius: '15px',
     animationName: [loaderAnimation],
     animationDuration: '.6s',
     animationIterationCount: 'infinite'
  },

  loader__line_1: {
     backgroundColor: '#FF5000',
     animationDelay: '.1s'
}
````

and so on

File: app/renderer/components/preferences/payment/enabledContent.js [hunk 6]
```diff
@@ -171,9 +170,23 @@ class EnabledContent extends ImmutableComponent {
...
+        <div className={css(styles.leader__wrap)}>
+          <div>
+            <div className={css(styles.loader__line, styles.loader__line_1, !inTranstion && styles.loader__line_off)} />
+            <div className={css(styles.loader__line, styles.loader__line_2, !inTranstion && styles.loader__line_off)} />
+            <div className={css(styles.loader__line, styles.loader__line_3, !inTranstion && styles.loader__line_off)} />
+          </div>
+        </div>
+      </div>
+      <div className={css(styles.enabledContent__walletBar)} data-test-id='walletBar'>
         <div className={css(gridStyles.row1col1, styles.walletBar__title)} data-l10n-id='monthlyBudget' />
```
Comment (luixxiul) [inline app/renderer/components/preferences/payment/enabledContent.js]: `enabledConent__walletBar__title`. please change every element under `enabledContent__walletBar`, thanks.

File: app/renderer/components/preferences/payment/enabledContent.js (hunk 1 above)
Comment (NejcZdovc) [reply app/renderer/components/preferences/payment/enabledContent.js]: done

File: app/renderer/components/preferences/payment/enabledContent.js (hunk 3 above)
Comment (luixxiul) [reply app/renderer/components/preferences/payment/enabledContent.js]: This change requires that: https://github.com/brave/browser-laptop/pull/11395/files#diff-5d8199641ea1ba12e0394859f7eb0c55R175

File: app/renderer/components/preferences/payment/enabledContent.js (hunk 6 above)
Comment (NejcZdovc) [reply app/renderer/components/preferences/payment/enabledContent.js]: done

Comment (luixxiul) [reply app/renderer/components/preferences/payment/enabledContent.js]: Sweet thanks!

File: app/browser/tabs.js [hunk 7]
```diff
@@ -198,6 +199,7 @@ const updateAboutDetails = (tab, tabValue) => {
       .merge(synopsis)
       .merge(preferencesData)
       .set('wizardData', wizardData)
+      .set('migration', migration)
```
Comment (bsclifton) [inline app/browser/tabs.js]: Love this! Great use of the new migration element

**Comment by bsclifton** [comment]
master https://github.com/brave/browser-laptop/commit/1d4edf4e628c9436627de267dc819da788536975
0.20.x https://github.com/brave/browser-laptop/commit/371ccf5d958cc050970a151f2e3ac3265ae41391
0.19.x https://github.com/brave/browser-laptop/commit/bb8be0da0a97e4d67dc15c604fd2b981f464baaa

@NejcZdovc since there were conflicts, would you mind checking my conflict resolution?  I manually tested the following on each:
- use fresh profile with 0.18.x branch
- create bitcoin wallet
- update branch
- run and confirm transition shows

**Pull request merged**